bliss/encoder/metrics.py

```diff
@@ -174,10 +174,64 @@ def compute(self):
         n_est_sources = self.n_est_sources[:-1] if self.exclude_last_bin else self.n_est_sources
         n_true_sources = self.n_true_sources[:-1] if self.exclude_last_bin else self.n_true_sources
 
+        precision_per_bin = (n_est_matches / n_est_sources).nan_to_num(0)
+        recall_per_bin = (n_true_matches / n_true_sources).nan_to_num(0)
+        f1_per_bin = (
+            2 * precision_per_bin * recall_per_bin / (precision_per_bin + recall_per_bin)
+        ).nan_to_num(0)
+
         precision = n_est_matches.sum() / n_est_sources.sum()
         recall = n_true_matches.sum() / n_true_sources.sum()
         f1 = 2 * precision * recall / (precision + recall)
 
+        if self.source_type_name == "total":
+            precision_bin_results = {
+                f"detection_precision_bin_{i}": precision_per_bin[i]
+                for i in range(len(precision_per_bin))
+            }
+            recall_bin_results = {
+                f"detection_recall_bin_{i}": recall_per_bin[i] for i in range(len(recall_per_bin))
+            }
+            f1_bin_results = {
+                f"detection_f1_bin_{i}": f1_per_bin[i] for i in range(len(f1_per_bin))
+            }
+
+            return {
+                "detection_precision": precision,
+                "detection_recall": recall,
+                "detection_f1": f1,
+                **precision_bin_results,
+                **recall_bin_results,
+                **f1_bin_results,
+            }
```

Comment:
Having a multiple return statements in a function is something of an anti-pattern. Please refactor this function to avoid duplicating code (the logic in the if block and afterwards seems about the same).

Reply:
fixed it